File: apps/api/src/modules/traces/trace.controller.ts
import { Controller, Get, Param } from '@nestjs/common';
import { TraceService } from './trace.service.js';

@Controller('traces')
export class TraceController {
  constructor(private readonly traceService: TraceService) {
    this.getTrace = this.getTrace.bind(this);
  }

  @Get(':traceId')
  async getTrace(@Param('traceId') traceId: string) {
    return this.traceService.getTraceDetails(traceId);
  }
}

File: apps/server-java/src/main/java/com/haoyitec/agent/server/module/trace/TraceController.java
package com.haoyitec.agent.server.module.trace;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/traces")
public class TraceController {

    private final TraceService traceService;

    @GetMapping("/{traceId}")
    public Object getTrace(@PathVariable String traceId) {
        return traceService.getTraceDetails(traceId);
    }
}

